pilot/pkg/config/kube/crdclient/types.gen.go

```diff
@@ -724,6 +730,83 @@ func patch(c kube.Client, orig config.Config, origMeta metav1.ObjectMeta, mod co
+			Spec:                         cfg.Spec.(*istioioapisecurityv1beta1.PeerAuthentication),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.ProxyConfig:
+		return c.Istio().NetworkingV1beta1().ProxyConfigs(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapinetworkingv1beta1.ProxyConfigApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapinetworkingv1beta1.ProxyConfig),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.RequestAuthentication:
+		return c.Istio().SecurityV1beta1().RequestAuthentications(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapisecurityv1beta1.RequestAuthenticationApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapisecurityv1beta1.RequestAuthentication),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.ServiceEntry:
+		return c.Istio().NetworkingV1alpha3().ServiceEntries(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapinetworkingv1alpha3.ServiceEntryApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapinetworkingv1alpha3.ServiceEntry),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.Sidecar:
+		return c.Istio().NetworkingV1alpha3().Sidecars(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapinetworkingv1alpha3.SidecarApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapinetworkingv1alpha3.Sidecar),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.Telemetry:
+		return c.Istio().TelemetryV1alpha1().Telemetries(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapitelemetryv1alpha1.TelemetryApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapitelemetryv1alpha1.Telemetry),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.VirtualService:
+		return c.Istio().NetworkingV1alpha3().VirtualServices(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapinetworkingv1alpha3.VirtualServiceApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapinetworkingv1alpha3.VirtualService),
+		}, metav1.ApplyOptions{FieldManager: fieldManager, Force: force})
+	case gvk.WasmPlugin:
+		return c.Istio().ExtensionsV1alpha1().WasmPlugins(cfg.Namespace).Apply(context.TODO(), &applyconfigistioioapiextensionsv1alpha1.WasmPluginApplyConfiguration{
+			ObjectMetaApplyConfiguration: getApplyConfigMetadata(objMeta),
+			Spec:                         cfg.Spec.(*istioioapiextensionsv1alpha1.WasmPlugin),
```

Comment:
Doesn't the spec need to be ApplyConfiguration as well?

Reply:
I think that might mean our client-go is broken.

Or our structs are unique relative to k8s core types so we don't need it possibly.

The reason its needed, btw, is to distinguish unset vs default. So I don't think we could apply something like `action=ALLOW` (default enum) for authz policy

Reply:
https://github.com/kubernetes-sigs/controller-tools/pull/536

Looks like this won't be possible using the k8s client for a bit.